### PR TITLE
ci: upgrade to otp 26

### PR DESCRIPTION
**Pull request by zmstone**
Fixes [EMQX-11510](https://emqx.atlassian.net/browse/EMQX-11510)
Release version: v/e5.4.0

## Summary
Build 5.4 on OTP 26.

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [ ] Added tests for the changes
- [ ] Added property-based tests for code which performs user input validation
- [ ] Changed lines covered in coverage report
- [ ] Change log has been added to `changes/(ce|ee)/(feat|perf|fix|breaking)-<PR-id>.en.md` files
- [ ] For internal contributor: there is a jira ticket to track this change
- [ ] Created PR to [emqx-docs](https://github.com/emqx/emqx-docs) if documentation update is required, or link to a follow-up jira ticket
- [ ] Schema changes are backward compatible

## Checklist for CI (.github/workflows) changes

- [ ] If changed package build workflow, pass [this action](https://github.com/emqx/emqx/actions/workflows/build_packages.yaml) (manual trigger)
- [ ] Change log has been added to `changes/` dir for user-facing artifacts update

### CODE REVIEW

**Review by ieQu1 — APPROVED** [review]

**Review by Rory-Z — APPROVED** [review]

File: apps/emqx/rebar.config [hunk 1]
```diff
@@ -66,7 +66,7 @@
     {plt_location, "."},
     {plt_prefix, "emqx_dialyzer"},
     {plt_apps, all_apps},
-    {plt_extra_apps, [hocon]},
+    {plt_extra_apps, [hocon,mnesia,bcrypt,os_mon,pbkdf2,emqx_http_lib, recon]},
```
Comment (keynslug) [inline apps/emqx/rebar.config:69]: Strange that we now need to do that, some of them probably need to be added to the `emqx.app.src`?

**Review by keynslug — APPROVED** [review]

File: apps/emqx/rebar.config (hunk 1 above)
Comment (zmstone) [reply apps/emqx/rebar.config:69]: mainly for two reasons
1. apps (e.g. mnesia) need to be added to all apps which calls mnesia, this is a bit annoying
2. apps (e.g. hocon, bcrpyt(when using nif), emqx_http_lib) they are static libs dummy root sup or no root sup at all.

Comment (keynslug) [reply apps/emqx/rebar.config:69]: Understood.

>  they are static libs dummy root sup or no root sup at all.

Even then, a lot of stuff treats entries in app.src as _dependencies that need to be taken into account_, not just dialyzer (i.e., relx).

Comment (zmstone) [reply apps/emqx/rebar.config:69]: Sorry. this is actually just for the standalone app.
Will fix it in the next PR.

Comment (zmstone) [reply apps/emqx/rebar.config:69]: and maybe bcrypt is not needed to be dynamic dep since we dropped windows.

Comment (zmstone) [reply apps/emqx/rebar.config:69]: https://github.com/emqx/emqx/pull/12179

**Pull request merged**